### ISSUE TITLE
Unknown OID for k256 ( 1.3.132.0.10 )

### ISSUE THREAD

**Issue by netshade** [RustCrypto/elliptic-curves#378]
Using the following dependencies:

```
[dependencies.elliptic-curve]
version="*"
features = ["pem", "arithmetic", "std", "zeroize", "pkcs8"]

[dependencies.k256]
version="*"
features=["ecdsa", "std", "pkcs8", "pem", "arithmetic"]
```

And the following trivial test:

```
#[test]
fn test_a_key() {
  use k256::pkcs8::FromPrivateKey;
  use k256::SecretKey;
  let key = "-----BEGIN PRIVATE KEY-----
A PRIVATE KEY WHOSE CONTENTS ARE NOT IN THIS MESSAGE
-----END PRIVATE KEY-----";
  let keypair = SecretKey::from_pkcs8_pem(key).unwrap();
}
```

I get the following panic:

```
test test_a_key ... FAILED

failures:

---- test_a_key stdout ----
thread 'test_a_key' panicked at 'called `Result::unwrap()` on an `Err` value: Asn1(Error { kind: UnknownOid { oid: ObjectIdentifier(1.3.132.0.10) }, position: None })', src/lib.rs:70:50
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I've tried tracing through the code ( I'm a Rust noob, so forgive me if I've missed obvious points ), and it's my understanding that the above is definitely not intended; that is to say, the `SecretKey` impl for `k256` specifies that the OID in question ( `1.3.132.0.10` ) should *always* be known. Is this a bug, or have I attempted to use `from_pkcs8_pem` incorrectly here?

**Comment by tarcieri** [comment]
The correct OID for secp256k1 is 1.3.132.0.10:

http://oid-info.com/get/1.3.132.0.10

The OID “1.3.132.0.1” is for the curve sect163k1:

http://oid-info.com/get/1.3.132.0.1

Whatever generated that key is using the wrong OID.

**Comment by netshade** [comment]
Bah, that is almost certainly the right curve for the key, and the fault is in my misreading the OID. Thanks for the reply, closing the comment.

**Comment by netshade** [comment]
Actually, scratch that - the fault is in my not pasting the entire OID, but the actual output OID is in fact `1.3.132.0.10`, as seen in the error message.

**Comment by netshade** [comment]
In the course of pasting another example, I actually found my problem here. You can ignore, thanks for taking the time to respond.

**Comment by tarcieri** [comment]
What was the problem?

**Comment by netshade** [comment]
I'm embarrassed to say that the key was in fact P256, rather than K256. I created a test key in OpenSSL that was K256 ( there were issues decoding that PEM as well, but I suspect that had to do with how I was creating it:

```
openssl genpkey -genparam -algorithm EC -pkeyopt ec_paramgen_curve:secp256k1 > sec.param
openssl genpkey -paramfile sec.param > testec.p8
```

) but since I was receiving an entirely different error, I prodded at the key and learned that the OID was for the p256 curve despite what I had initially thought. Operator error. :)
